File: 백준/Bronze/1032. 명령 프롬프트/명령 프롬프트.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const n = Number(input[0]);
const files = input.slice(1, n + 1);

let pattern = '';
const length = files[0].length;

for (let i = 0; i < length; i++) {
  const char = files[0][i];
  const allSame = files.every(file => file[i] === char);
  pattern += allSame ? char : '?';
}

console.log(pattern);

File: 백준/Bronze/1032. 명령 프롬프트/명령 프롬프트.py
n = int(input())
files = [input().rstrip() for _ in range(n)]

length = len(files[0])
pattern = []

for i in range(length):
    char = files[0][i]
    all_same = all(file[i] == char for file in files)
    pattern.append(char if all_same else '?')

print(''.join(pattern))
